### PR TITLE
fix(DB/Trainer) Add Flying Carpet to Tailoring trainers

### PR DESCRIPTION
**Pull request by noisiver**
## Changes Proposed:
-  Add Flying Carpet to the tailoring trainers that were missing it

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15157

## SOURCE:

https://www.wowhead.com/wotlk/spell=60969/flying-carpet#taught-by-npc

## Tests Performed:

- Checked the listed creatures to make sure they have the spell available in their list

## How to Test the Changes:

1. Check the creatures listed in the SQL and make sure they have the spell available to train

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
Correct me if I'm wrong, but wouldn't this also add 67 other spells to those trainers?

image

**Comment by noisiver** [comment]
> Correct me if I'm wrong, but wouldn't this also add 67 other spells to those trainers?
> 
> image

Oh, right. I would have to confirm all those before adding them. Sigh. I don't know if adding the spell straight up is a viable option but I'll look at it later.

**Comment by Gultask** [comment]
It could very well be the references that are wrong, and you'd only need to add that spell to a reference they all share

**Comment by noisiver** [comment]
I'll take a look at it as soon as I can. Thank you for reminding me about this since I didn't even consider it.

**Comment by noisiver** [comment]
Took me long enough but I updated it to only add Flying Carpet to said trainers.

**Comment by Maelthyrr** [comment]
Hi, Thanks for the PR. Look to work well.

**Comment by Gultask** [comment]
Wouldn't it be better to remove it from reference 201041 and add it to reference 201040 instead?

Spell 60969 is currently in reference 201041
image

**Comment by noisiver** [comment]
> 

It depends on if that reference is used by all trainers required and also no other trainers. I did look on wowhead, the wotlk classic database, and it does have the list of trainers on it so I guess I could look into it.

**Comment by Gultask** [comment]
I think everyone's here, no?
ID 26964 uses reference 201040
image

**Comment by noisiver** [comment]
I was confusing myself but there is in fact one trainer that **should** have Flying Carpet but doesn't have that reference. What should I do about that?

image
image

The first image is all trainers that has that reference and the second is all that should have Flying Carpet. [Meline Duskpath](https://www.wowhead.com/wotlk/npc=33596/meline-duskpath) is the one that should have it but doesn't have that reference.

**Comment by noisiver** [comment]
For now I've added it to the reference. Hopefully you can give me some advice on how to handle the last trainer. Perhaps it's missing the reference.

File: data/sql/updates/pending_db_world/rev_1677089599224449400.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+--
+DELETE FROM `npc_trainer` WHERE `SpellID`=60969;
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1677089599224449400.sql:2]: Usually it's better to add 2 fields when deleting. Just to be a bit safer
```suggestion
DELETE FROM `npc_trainer` WHERE `SpellID`=60969 AND `ID` IN (201040, 201041);
```

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**